job/job_test.go

```diff
@@ -173,3 +176,55 @@ func TestSetupFails(t *testing.T) {
 	// should still be able to log, see #_incident-130
 	j.With("happyThoughts", true).Error(err)
 }
+
+type mockJob struct{}
```

Comment:
All this new test code can be split off into a separate test func and then we can easily make it a table-driven test by having the `tw.WorkflowContextList` be part of the table. Another reason to make `isHardwareHavingActiveWorflow` take in the `tw.WorkflowContextList` instead of a fetcher.

Reply:
:) in that case we will have to add separate unit test for fetcher which will use existing structs of type mocks to complete the testing. I mean in that, `isHardwareHavingActiveWorflow` will lose it sole purpose and having it unit tested will be minuscule compared to fetcher method.

Reply:
added tabledriven test